internal/service/alarms/internal/notifier_provider/subscription.go

```diff
@@ -48,8 +49,16 @@ func (s *SubscriptionStorageProvider) GetSubscriptions(ctx context.Context) ([]n
 }
 
 func (s *SubscriptionStorageProvider) Matches(subscription *notifier.SubscriptionInfo, notification *notifier.Notification) bool {
-	// TODO we to need somehow group notifications per subscriber in slight nicer way, see GetNotifications for more
-	return subscription.SubscriptionID == notification.NotificationID
+	payload, ok := notification.Payload.(generated.AlarmEventNotification)
+	if !ok {
+		slog.Warn("notification payload is not of type generated.AlarmEventNotification", "type", fmt.Sprintf("%T", notification.Payload))
+		return false
+	}
+	if subscription.Filter == nil {
+		return true
+	}
+	filter := generated.AlarmSubscriptionInfoFilter(*subscription.Filter)
+	return models.AlarmFilterToEventType(filter) == payload.NotificationEventType
```

Comment:
its such strange thing but the spec says to do the opposite. 

```
        filter:
          type: string
          enum: [ NEW, CHANGE, CLEAR, ACKNOWLEDGE ]
          description: |
            Criteria for events which do not need to be reported or will be filtered by the subscription 
            notification service. Therefore, if a filter is not provided then all events are reported.
            It can be filtered by criteria based on the type of notification of fields of the
            AlarmEventRecord.
          example: NEW
```

it matches if `!=`

Reply:
😮  ... fixed